src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -0,0 +1,219 @@
+    queueFifoTimeSpanMins = timeSpanMins;
+  }
+
+  /**
+   * Convert a long time pointer to format required for EDB calls.
+   *
+   * @param time long time pointer to format
+   * @return formatted time string for EBD calls
+   */
+  private static String convertToEBDTimeFormat(long time) {
+    return new SimpleDateFormat(HistoricalDataConstants.EBD_TIME_FORMAT).format(new Date(time));
+  }
+
+  /**
+   * Get the current long time pointer.
+   *
+   * @return current time as long
+   */
+  private static long getCurrentTime() {
+    return System.currentTimeMillis();
+  }
+
+  /**
+   * Get the historical log data for all tag groups within the next FIFO queue time span.
+   *
+   * @param startNewTimeTracker if new time tracker should be generated, not read from storage
+   * @return historical log data
+   * @throws IOException if unable to read or write files
+   */
+  public static synchronized ArrayList getFifoNextSpanDataAllGroups(boolean startNewTimeTracker)
+      throws IOException {
+    final boolean includeTagGroupA = true;
+    final boolean includeTagGroupB = true;
+    final boolean includeTagGroupC = true;
+    final boolean includeTagGroupD = true;
+    return getFifoNextSpanData(
+        startNewTimeTracker,
+        includeTagGroupA,
+        includeTagGroupB,
+        includeTagGroupC,
+        includeTagGroupD);
+  }
+
+  /**
+   * Get the historical log data for the specified tag groups within the next FIFO queue time span.
+   *
+   * @param startNewTimeTracker if new time tracker should be generated, not read from storage
+   * @param includeTagGroupA if tag group A data should be included
+   * @param includeTagGroupB if tag group B data should be included
+   * @param includeTagGroupC if tag group C data should be included
+   * @param includeTagGroupD if tag group D data should be included
+   * @return historical log data
+   * @throws IOException if unable to read or write files
+   */
+  public static synchronized ArrayList getFifoNextSpanData(
+      boolean startNewTimeTracker,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD)
+      throws IOException {
+    // Get start time from file, or start new time pointer if startNewTimePtr is true.
+    final String timeMarkerFileName =
+        HistoricalDataConstants.QUEUE_FILE_FOLDER
+            + HistoricalDataConstants.QUEUE_TIME_FILE_NAME
+            + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
+    String startTimeTracker = "";
+    if (startNewTimeTracker) {
+      startTimeTracker = "" + getCurrentTime();
+      writeDataFile(timeMarkerFileName, startTimeTracker);
+    } else {
+      startTimeTracker = readDataFile(HistoricalDataConstants.FILE_URL_PREFIX + timeMarkerFileName);
+    }
+    final long startTimeTrackerLong = Long.parseLong(startTimeTracker);
+
+    // Calculate end time with FIFO time span
+    String endTimeTracker = "";
```

Comment:
Long.toString() should be used here as im pretty sure its getting called anyway. Then you dont have to have the flexy do the string concat